saleor/graphql/decorators.py

```diff
@@ -23,15 +23,30 @@ def wrapper(context, *args, **kwargs):
 
 def permission_required(perm):
     def check_perms(context):
-        if isinstance(perm, six.string_types):
-            perms = (perm,)
-        else:
-            perms = perm
-        if context.user.has_perms(perms):
-            return True
-        service_account = getattr(context, "service_account", None)
-        if service_account and service_account.has_perms(perms):
-            return True
+        return _permission_required(perm, context)
+
+    return account_passes_test(check_perms)
+
+
+def one_of_permissions_required(perms):
+    def check_perms(context):
+        for perm in perms:
+            has_perm = _permission_required(perm, context)
+            if has_perm:
+                return True
         return False
 
     return account_passes_test(check_perms)
+
+
+def _permission_required(perm, context):
+    if isinstance(perm, six.string_types):
```

Comment:
I think we should only do this in `permission_required` and `one_of_permissions_required` should expect a list (with proper type hinting)